### ISSUE TITLE
Tracer crash windup with delay slot

### ISSUE THREAD

**Issue by sei-eschwartz** [angr/angr#3155]
---

**Describe the bug.**

The final step of tracer's [crash windup](https://sourcegraph.com/github.com/angr/angr/-/blob/angr/exploration_techniques/tracer.py?L955) is to step by one instruction.  However, on architectures with delay slots this may be problematic because of #71.  Here's an interactive debug session showing the issue:

```
(Pdb) p state.block().pp()
0x40084c:	jr	$ra
0x400850:	nop	
None
(Pdb) p state.block()
<Block for 0x40084c, 8 bytes>
(Pdb) p state.step(num_inst=1)
*** angr.errors.SimIRSBNoDecodeError: IR decoding error at 0x40084c. You can hook this instruction with a python replacement using project.hook(0x40084c, your_function, length=length_of_instruction).
(Pdb) p state.step(num_inst=2)
WARNING:angr.engines.successors:Exit state has over 256 possible solutions. Likely unconstrained; skipping. <BV32 packet_0_stdin_sps_11_8192[7111:7104] .. packet_0_stdin_sps_11_8192[7119:7112] .. packet_0_stdin_sps_11_8192[7127:7120] .. packet_0_stdin_sps_11_8192[7135:7128]>
INFO:angr.engines.engine:Ticked state: <IRSB from 0x40084c: 1 unconstrained>
<IRSB from 0x40084c: 1 unconstrained>
```

**Environment Information.**

```
(angr) ed@ed-Precision-7540:~/Documents/Angry-CodeReuse$ python -m angr.misc.bug_report
angr environment report
=============================
Date: 2022-03-03 11:37:04.180925
Running in virtual environment at /home/ed/.virtualenvs/angr
Platform: linux-x86_64
Python version: 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0]
######## angr #########
Python found it in /home/ed/Documents/angr-dev/angr/angr
Pip version angr 9.0.gitrolling
Git info:
	Current commit 689a748180d4177cc822c3fac599ef0c83768e8a from branch vault_test
	Checked out from remote ed: https://edmcman@github.com/edmcman/angr
######## ailment #########
Python found it in /home/ed/Documents/angr-dev/ailment/ailment
Pip version ailment 9.0.gitrolling
Git info:
	Current commit 1650314856cd8ffdc6e34672fd9722ee05adceb0 from branch master
	Checked out from remote origin: https://github.com/angr/ailment
######## cle #########
Python found it in /home/ed/Documents/angr-dev/cle/cle
Pip version cle 9.0.gitrolling
Git info:
	Current commit 9550f76a19fb717c168f1d3495498ddf753373c2 from branch master
	Checked out from remote origin: https://github.com/angr/cle
######## pyvex #########
Python found it in /home/ed/Documents/angr-dev/pyvex/pyvex
Pip version pyvex 9.0.gitrolling
Git info:
	Current commit 8cf6a994c32d4d53cbfe99e77c1bc629377c6978 from branch master
	Checked out from remote origin: https://github.com/angr/pyvex
######## claripy #########
Python found it in /home/ed/Documents/angr-dev/claripy/claripy
Pip version claripy 9.0.gitrolling
Git info:
	Current commit d53647bfa0ee6c3e4c1c794c0b3beba2edbdf8c7 from branch master
	Checked out from remote origin: https://github.com/angr/claripy
######## archinfo #########
Python found it in /home/ed/Documents/angr-dev/archinfo/archinfo
Pip version archinfo 9.0.gitrolling
Git info:
	Current commit 1ea999a9ae4db005a1db8e76bd5b67769850e8e8 from branch master
	Checked out from remote origin: https://github.com/angr/archinfo
######## z3 #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/z3
Pip version z3-solver 4.8.7.0
Couldn't find git info
######## unicorn #########
Python found it in /home/ed/.virtualenvs/angr/lib/python3.6/site-packages/unicorn
Pip version unicorn 1.0.2rc4
Couldn't find git info
######### Native Module Info ##########
angr: <CDLL '/home/ed/Documents/angr-dev/angr/angr/state_plugins/../lib/angr_native.so', handle 273cc60 at 0x7f3337ee9ac8>
unicorn: <CDLL '/home/ed/.virtualenvs/angr/lib/python3.6/site-packages/unicorn/lib/libunicorn.so', handle 18f7240 at 0x7f336dfdff60>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7f336f00e4a8>
z3: <CDLL '/home/ed/.virtualenvs/angr/lib/python3.6/site-packages/z3/lib/libz3.so', handle 1cc63c0 at 0x7f336a155cc0>
```

**To Reproduce.**

I don't have this right now but I can provide it if necessary.

**Additional context.**

I'm not sure what the fix for this should be.  Try to step by one instruction, and if we get a lifting error, try two instructions?

**Comment by rhelmot** [comment]
I like the hack solution. Feel free to implement it!

**Comment by sei-eschwartz** [comment]
Fixed in #3177
